src/bin/wasm-tools/component.rs

```diff
@@ -366,6 +369,66 @@ impl WitOpts {
     }
 }
 
+/// Tool for verifying whether a component conforms to a world.
+#[derive(Parser)]
+pub struct TargetsOpts {
+    #[clap(flatten)]
+    general: wasm_tools::GeneralOpts,
+    /// The WIT package containing the `world` used to test a component for conformance.
+    ///
+    /// This can either be a directory or a path to a single `*.wit` file.
+    wit: PathBuf,
+    /// The world used to test whether a component conforms to its signature.
+    #[clap(short, long)]
+    world: Option<String>,
+    /// Input file or directory to process.
+    ///
+    /// The file specified can be a `*.wit` file parsed as a single-document
+    /// package. It can be a directory to be parsed as a WIT package. It can be
+    /// a `*.wat` or `*.wasm` file for either the binary representation of a WIT
+    /// package or a component itself to extract the interface from. The type of
+    /// input is inferred from the contents of the path specified.
+    ///
+    /// If not provided or if this is `-` then stdin is read entirely and
+    /// processed.
+    input: Option<PathBuf>,
```

Comment:
Since the input here is wasm then this can be `io: wasm_tools::InputOutput` which covers the text format, stdin, etc. (there's some examples of this throughout the other subcommands, the `input` here from`WitOpts` above is a bit special because the input to that command isn't necessarily wasm unlike many others)

Reply:
I had considered this; I avoided using `InputOutput` because it introduced an `-o/--output` option which doesn't seem to make much sense in this case. Nevertheless, happy to use the `InputOutput` type if that's the convention for input-only as well!

Reply:
Oh true, forgot about that! If you're up for it it might be good to split `InputOutput` into an `Input` and an `Output` (but still have the fused `InputOutput`). Mostly just trying to keep the logic for parsing input consolidated to ensure consistent behavior across subcommands.

Reply:
Perfect, totally agree! I started down this path already should have an update soon. Thanks!

Reply:
Addressed this in #1099